### PR TITLE
Launch reports when saving or resetting validation

### LINKED ISSUE #208: Report logic not triggered when validated entries are saved
The logic for creating reports is not triggered when data is validated and the entries are saved. This means there is not data to plot when trying to generate an Hourly, Daily or Monthly report.

### PR DESCRIPTION
**Pull request by dalonsoa**
Launches the calculation of the reports when saving the validated data, either when saving days or when saving individual entries. It also deletes reports data in the relevant time range when resetting validation.

Fixes #208

### CODE REVIEW

File: measurement/validation.py [hunk 1]
```diff
@@ -277,33 +287,54 @@ def reset_validated_entries(ids: list) -> None:
     Args:
         ids (list): List of measurement ids to reset.
     """
+    times = []
     update = {"is_validated": False, "is_active": True}
     for _id in ids:
-        Measurement.objects.filter(id=_id).update(**update)
+        current = Measurement.objects.filter(id=_id)
+        current.update(**update)
+        times.append(current.time)
```
Comment (tsmbland) [inline measurement/validation.py:295]: I get `AttributeError: 'QuerySet' object has no attribute 'time'`

File: measurement/validation.py [hunk 2]
```diff
@@ -277,33 +287,54 @@ def reset_validated_entries(ids: list) -> None:
     Args:
         ids (list): List of measurement ids to reset.
     """
+    times = []
     update = {"is_validated": False, "is_active": True}
     for _id in ids:
-        Measurement.objects.filter(id=_id).update(**update)
+        current = Measurement.objects.filter(id=_id)
```
Comment (tsmbland) [inline measurement/validation.py:293]:
```suggestion
        current = Measurement.objects.get(id=_id)
        current.is_validated = False
        current.is_active = True
        current.save()
```

Something like this?

**Review by tsmbland — CHANGES_REQUESTED** [review]
Mostly looks good, but the reset validation button for the detail table isn't working for the reason stated below

File: measurement/validation.py (hunk 2 above)
Comment (dalonsoa) [reply measurement/validation.py:293]: Yes, this should do it. It's funny, I was not getting any error when trying the function...

**Comment by tsmbland** [comment]
Ok now a test is failing. Not sure what's going on though as only the `reset_validated_entries` function was changed and that function isn't actually used anywhere in the test suite...

**Comment by dalonsoa** [comment]
I don't know what's going on. I cannot reproduce it locally. It works well in my laptop. Can you try in your computer?

**Comment by dalonsoa** [comment]
After merging @CWestICL PR and updating this branch, tests are passing, so no idea what's going on...

**Review by tsmbland — APPROVED** [review]

**Review by CWestICL — APPROVED** [review]
Seems to be working for me!

**Pull request merged**